packages/components/src/SubHeaderBar/SubHeaderBar.component.js

```diff
@@ -7,85 +8,130 @@ import { DEFAULT_I18N, getDefaultTranslate } from '../translate';
 import { Action } from '../Actions';
 import ActionBar from '../ActionBar';
 import InputTitleSubHeader from './InputTitleSubHeader';
+import Inject from '../Inject';
 import theme from './SubHeaderBar.scss';
 
-function SubHeaderBarActions({ components, left, right, center, className }) {
+function SubHeaderBarActions({ children, tag, left, right, center, hasRight }) {
+	const className = classNames({
+		'subheader-left': left,
+		'subheader-center': center,
+		'subheader-right': right,
+		[theme['tc-subheader-navbar-left']]: left,
+		[theme['tc-subheader-navbar-right']]: right,
+		[theme['tc-subheader-navbar-center']]: center,
+		[theme['tc-subheader-navbar-center-no-margin-right']]: hasRight,
+		'tc-subheader-navbar-center-no-margin-right': hasRight,
+	});
 	return (
 		<div className={className}>
-			{components.map((component, index) => (
-				<ActionBar.Content
-					key={index}
-					tag={component.tag}
-					left={left}
-					center={center}
-					right={right}
-				>
-					{component.injectedComponent}
-				</ActionBar.Content>
-			))}
+			<ActionBar.Content
+				tag={tag}
+				left={left}
+				center={center}
+				right={right}
+			>
+				{children}
+			</ActionBar.Content>
 		</div>
 	);
 }
 
 SubHeaderBarActions.propTypes = {
-	components: PropTypes.array.isRequired,
+	children: PropTypes.node,
+	tag: PropTypes.string,
 	right: PropTypes.bool,
 	center: PropTypes.bool,
 	left: PropTypes.bool,
-	className: PropTypes.string,
+	hasRight: PropTypes.bool,
 };
 
-function SubHeaderBar({ t, onGoBack, componentsCenter, componentsRight, className, ...rest }) {
-	const defaultComponentsLeft = [
-		{
-			injectedComponent: (
-				<Action
-					id="backArrow"
-					onClick={onGoBack}
-					label={t('BACK_ARROW_TOOLTIP', { defaultValue: 'Go Back' })}
-					icon="talend-arrow-left"
-					bsStyle="link"
-					className={classNames(theme['tc-subheader-back-button'], 'tc-subheader-back-button')}
-					hideLabel
-				/>
-			),
-		},
-		{
-			injectedComponent: <InputTitleSubHeader t={t} {...rest} />,
-		},
-	];
+function CustomInject({ getComponent, left, right, center, nowrap, ...props }) {
+	if (nowrap) {
+		return <Inject getComponent={getComponent} {...props} />;
+	}
+	return (
+		<SubHeaderBarActions left={left} right={right} center={center}>
+			<Inject getComponent={getComponent} {...props} />
+		</SubHeaderBarActions>
+	);
+}
+CustomInject.propTypes = {
+	nowrap: PropTypes.bool,
+	right: PropTypes.bool,
+	center: PropTypes.bool,
+	left: PropTypes.bool,
+	getComponent: PropTypes.func,
+};
+
+function SubHeaderBar({
+	t,
+	onGoBack,
+	components,
+	getComponent,
+	className,
+	children,
+	left,
+	center,
+	right,
+	...rest
+}) {
+	const injected = Inject.all(getComponent, components, CustomInject);
+	const Components = Inject.getAll(getComponent, { Action, ActionBar });
```

Comment:
Using `Components` and `components` in the same component with a single case difference is misleading. Please find a better name.

Reply:
Hope you prefer Renderer